fury/ui/elements.py

```diff
@@ -3272,3 +3272,696 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+        _node = self.nodes_dict[node]
+        return _node
+
+    @property
+    def nodes(self):
+        """Get all the nodes present in the Tree2D
+
+        Returns
+        -------
+        list of nodes
+        """
+        return self._nodes
+
+    @property
+    def nodes_dict(self):
+        """Get all the nodes present in the Tree2D in dict format
+
+        Returns
+        -------
+        dict with label, node as key, value
+        """
+        return self._nodes_dict
+
+
+class TreeNode2D(UI):
+    """Node/Leaf of a Tree2D UI
+
+    Attributes
+    ----------
+    children: list of :class: `TreeNode2D`
+        Sub nodes of the current node
+    parent: :class: `TreeNode2D`
+        Parent node of the current node
+    indent: int
+        Indentaion of the current node
+    label: str
+        Label text of the current node
+    expanded: bool
+        Whether the current node is expanded or not
+    """
+
+    def __init__(self, label, parent=None, children=None, icon=None,
+                 position=(0, 0), size=(200, 200), indent=20,
+                 child_indent=10, child_height=25, color=(0.3, 0.3, 0.3),
+                 opacity=0.8, expandable=True, expanded=False,
+                 selected_color=(0.8, 0.3, 0.3), auto_resize=True):
+        """Initialize the UI element
+
+        Parameters
+        ----------
+        label: str
+            Label text of the current node
+        children: list of :class: `TreeNode2D`, optional
+            Sub nodes of the current node
+        icon: str, optional
+            Path/URl to the icon placed next to the label
+        parent: :class: `TreeNode2D`, optional
+            Parent node of the current node
+        position : (float, float), optional
+            Absolute coordinates (x, y) of the lower-left corner of the
+            UI component
+        size : (int, int), optional
+            Width and height of the pixels of this UI component.
+        indent: int, optional
+            Indentation of the current node
+        child_indent: int, optional
+            Indentation of the child nodes
+        child_height: int, optional
+            Space taken by each sub-node vertically
+        color : list of 3 floats, optional
+            Background color of current node.
+        opacity: float, optional
+            Background opacity of the current node
+        expandable: bool, optional
+            If the node should expand/collapse
+        expanded: bool, optional
+            Whether the current node is expanded or not
+        selected_color: list of 3 floats, optional
+            Color of the selected node.
+        unselected_color: list of 3 floats, optional
+            Color of the unselected node.
+        auto_resize: bool, optional
+            If the node should automatically resize to fit its content.
+        """
+        self.children = children
+        self.icon = icon
+        if self.children is None:
+            self.children = []
```

Comment:
`self.children = children or []` to make it simpler

Reply:
Done.